server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -499,10 +500,12 @@ public void validateMapping(Metadata metadata, final IndexMetadata newIndexMetad
             return;
         }
         var tableInfo = tableFactory.create(RelationName.fromIndexName(indexName), metadata);
-        this.getTranslogIndexer = () -> new TranslogIndexer(tableInfo);
+        var indexer = new TranslogIndexer(tableInfo);
+        this.getTranslogIndexer = () -> indexer;
```

Comment:
Is this still fine in regards to the race that was fixed in https://github.com/crate/crate/commit/906d468230d6950a2a706910459eef81451551fc ?

Reply:
It should be.  The issue there was that the TableInfo could get out of sync, and we're creating that in the same place and persisting it for the same duration; the difference now is that we also create the TranslogIndexer from it up-front, rather than every time `.get()` is called